packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -2039,6 +2046,854 @@ contract OPContractsManagerInteropMigrator is OPContractsManagerBase {
+                string memory l1XdmName = "OVM_L1CrossDomainMessenger";
+                result = Blueprint.deployFrom(
+                    bps.resolvedDelegateProxy,
+                    computeSalt(_args.l2ChainId, _args.saltMixer, "L1CrossDomainMessenger"),
+                    abi.encode(_args.addressManager, l1XdmName)
+                );
+
+                // ResolvedDelegateProxy requires setting the proxy type on the ProxyAdmin.
+                _args.proxyAdmin.setProxyType(result, IProxyAdmin.ProxyType.RESOLVED);
+                _args.proxyAdmin.setImplementationName(result, l1XdmName);
+            } else {
+                // Otherwise this is a normal proxy.
+                result = deployProxy(_args.l2ChainId, _args.proxyAdmin, _args.saltMixer, _contractName);
+            }
+
+            // Emit the proxy creation event.
+            emit ProxyCreation(_contractName, result);
+
+            // Return the final deployment result.
+            return result;
+        }
+    }
+
+    /// @notice Resets the initialized slot for a contract and then initializes it.
+    /// @param _proxyAdmin The proxy admin of the contract.
+    /// @param _target The target of the contract.
+    /// @param _implementation The implementation of the contract.
+    /// @param _data The data to call the initializer with.
+    function _resetAndInitialize(
+        IProxyAdmin _proxyAdmin,
+        address _target,
+        address _implementation,
+        bytes memory _data
+    )
+        internal
+    {
+        _resetAndInitialize(_proxyAdmin, _target, _implementation, _data, bytes32(0), 0);
+    }
+
+    /// @notice Resets the initialized slot for a contract and then initializes it.
+    /// @param _proxyAdmin The proxy admin of the contract.
+    /// @param _target The target of the contract.
+    /// @param _implementation The implementation of the contract.
+    /// @param _data The data to call the initializer with.
+    /// @param _slot The slot where the initialized value is located.
+    /// @param _offset The offset of the initializer value in the slot.
+    function _resetAndInitialize(
+        IProxyAdmin _proxyAdmin,
+        address _target,
+        address _implementation,
+        bytes memory _data,
+        bytes32 _slot,
+        uint8 _offset
+    )
+        internal
+    {
+        // Upgrade to StorageSetter.
+        _proxyAdmin.upgrade(payable(_target), address(implementations().storageSetterImpl));
+
+        // Reset the initialized slot by zeroing the single byte at `_offset` (from the right).
+        bytes32 current = StorageSetter(_target).getBytes32(_slot);
+        uint256 mask = ~(uint256(0xff) << (uint256(_offset) * 8));
+        StorageSetter(_target).setBytes32(_slot, bytes32(uint256(current) & mask));
+
+        // Upgrade to the implementation and call the initializer.
+        _proxyAdmin.upgradeAndCall(payable(address(_target)), _implementation, _data);
```

Comment:
I wish there were initializer getter methods that `Initializable` has so we can assert onchain that we properly reset the initializer and did not change anything else, but seems those don't exist